### PR TITLE
chore: bump manifests 2.0.3 -> 2.2.0

### LINKED ISSUE #95: Update `envoy` manifests
### Context

Each charm has a set of manifest files that have to be upgraded to their target version. The process of upgrading manifest files usually means going to the component’s upstream repository, comparing the charm’s manifest against the one in the repository and adding the missing bits in the charm’s manifest.

### What needs to get done

https://docs.google.com/document/d/1a4obWw98U_Ndx-ZKRoojLf4Cym8tFb_2S7dq5dtRQqs/edit?pli=1#heading=h.jt5e3qx0jypg

### Definition of Done

1. Manifests are updated 
2. Upstream image is used

### PR DESCRIPTION
**Pull request by orfeas-k**
* Update manifests and envoy configuration
* Add CONTRIBUTING.md file with instructions for updating manifests

#### Attention, this PR removes an integration test
I document in detail in #106 the situation with integration tests but summary is that envoy is functional and KFP pipeline steps can get the artifacts while curling envoy with the same headers fails. I filed an issue so we can work on envoy integration tests separately, since we need to spend more time on understanding envoy - kfp-ui relation and imitate this in our tests. I 've already done some exploration but did not want to spend more time here, since this has already been overdue and I would not want to introduce a test that I do not understand. Instructions for manual testing are below.  

#### Testing instructions
1. Deploy CKF from latest/edge using juju agent 3.5.0 due to https://github.com/canonical/bundle-kubeflow/issues/921
2. Refresh to this PR's charm
  ```
  j refresh envoy --channel latest/edge/pr-102 --resource oci-image=gcr.io/ml-pipeline/metadata-envoy:2.2.0
  ```
3. Start an experiment and observe that their runs steps are getting artifacts while there is no error in the browser dev tools network tab.

Closes #95

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Update manifests
+
+Envoy use a Jinja2 template to store its `envoy.yaml` and applies during its deployments. This is stored in `src/templates`. Envoy-related manifests are part of the KFP manifests. The process for updating them is:
+
+### Spot the differences between versions of manifest files
+
+1. Install `kustomize` using the official documentation [instructions](https://kubectl.docs.kubernetes.io/installation/kustomize/)
+2. Clone [Kubeflow manifests](https://github.com/kubeflow/manifests) repo locally
+3. `cd` into the repo and checkout to the branch or tag of the target version.
+4. Build the manifests with `kustomize` according to instructions in https://github.com/kubeflow/manifests?tab=readme-ov-file#kubeflow-pipelines.
```
Comment (DnPlas) [inline CONTRIBUTING.md:10]: AFAIK the manifests for metadata envoy are not rendered from this command. Could you please confirm the source is correct?

**Review by DnPlas — COMMENTED** [review]
Thanks @orfeas-k, tiny little comment, other than that lgtm.

**Review by DnPlas — APPROVED** [review]
Thanks @orfeas-k !

**Review by misohu — APPROVED** [review]
Approving ... I tried the deployment from PR description without problems .... maybe we should label the integration test fix issue for 1.9 WDYT ?

**Pull request merged**